server/src/machines/dre/api.rs

```diff
@@ -0,0 +1,102 @@
+use std::time::Duration;
+use crate::serial::devices::dre::DreData;
+use super::DreMachine;
+use control_core::{
+    machines::api::MachineApi,
+    socketio::{
+        event::{Event, GenericEvent},
+        namespace::{
+            CacheFn, CacheableEvents, Namespace, NamespaceCacheingLogic, NamespaceInterface,
+            cache_duration,
+        },
+    },
+};
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use uom::si::f64::Length;
+
+
+#[derive(Serialize, Debug, Clone)]
+pub struct DiameterEvent {
+    pub dre_data: Option<DreData>,
+}
+
+impl DiameterEvent {
+    pub fn build(&self) -> Event<Self> {
+        Event::new("DiameterResponseEvent", self.clone())
+    }
+}
+
+pub enum DreEvents {
+    DiameterEvent(Event<DiameterEvent>),
+}
+
+#[derive(Debug)]
+pub struct DreMachineNamespace(Namespace);
+
+impl DreMachineNamespace {
+    pub fn new() -> Self {
+        Self(Namespace::new())
+    }
+}
+
+impl CacheableEvents<DreEvents> for DreEvents {
+    fn event_value(&self) -> Result<GenericEvent, serde_json::Error> {
+        match self {
+            DreEvents::DiameterEvent(event) => event.try_into(),
+        }
+    }
+
+    fn event_cache_fn(&self) -> CacheFn {
+        let cache_ten_secs = cache_duration(Duration::from_secs(10));
+
+        match self {
+            DreEvents::DiameterEvent(_) => cache_ten_secs,
+        }
+    }
+}
+
+
+#[derive(Deserialize, Serialize)]
+enum Mutation {
+    TargetSetTargetDiameter(Length),
+    TargetSetLowerTolerance(Length),
+    TargetSetHigherTolerance(Length)
+}
```

Comment:
Don't use `Length` in the interface, just use an f64 and comment that the `/// Unit is mm`

Please clarify if the lowe tolerance should be a negative or positive value.

I would prefer a negative value